report/src/sections/07-future-works.tex

```diff
@@ -7,8 +7,18 @@ \section{Future Works}\label{sec:future-works}
 The extension of the abstract interpretation to support floating point numbers would allow for a more precise analysis of the database queries.
 One of the obstacles to this extension is the fact that floating point numbers are represented in a different ways in different programming languages, which makes it difficult to create a general abstract interpretation for floating point numbers.
 
-Another area that could be further developed is the precision of the semantic in som cases.
-The current set of semantics are very similar to the ones fund in the original paper by Halder and Cortesi~\cite{halder_abstract_2012}, and they are not very precise.
-This could then lead to a more precise analysis of the database area and more accurate results when judging the abstract interpretation of the database.
+Another area that could be further developed is the rigorousness of the soundness proof.
+At the current state of the soundness proof is more akin to a sketch of a proof, rather than a full proof.
+Therefore, one of the future works could be to further develop the soundness proof to make it more rigorous.
+This would involve going through the proof and making sure that all the steps are correct and that all the assumptions are valid, which would require a lot of time and effort, but would make the proof more reliable and increase the confidence in the results of the abstract interpretation.
 
-Currently, this paper only shows the theoretical approach to this problem therefore one of the future works could be to implement the abstract semantics in a program that could be used to analyze a database system and check for some requested properties.
\ No newline at end of file
+Besides the rigor, you could also try to increase the precision of the abstract interpretation.
+This could be done by adding more rules to the abstract interpretation, such as a better way of handling Boolean values and how to more accurately represent them, which would allow for a more precise analysis of the database queries.
```

Comment:
I think what we should say here instead is that a proof of optimality should be carried out, but that we are not sure the current analysis is optimal at the moment.